cli/src/config.rs

```diff
@@ -183,25 +183,29 @@ impl WithPath<Config> {
             .members
             .iter()
             .map(|m| {
-                self.path()
-                    .parent()
-                    .unwrap()
-                    .join(m)
-                    .canonicalize()
-                    .unwrap()
+                self.path().parent().unwrap().join(m).canonicalize().expect(
+                    format!(
+                        "Error reading workspace.members. \
+                         File {:?} does not exist at path {:?}.",
+                        m, self.path
+                    )
+                    .as_str(),
+                )
```

Comment:
```suggestion
                self.path().parent().unwrap().join(m).canonicalize().unwrap_or_else(|_|
                    panic!(
                        "Error reading workspace.members. \
                         File {:?} does not exist at path {:?}.",
                        m, self.path
                    )
                )
```
`clippy` will not like `expect` with `format!`. Same for other `expect`.

Reply:
@fanatid What is your suggestion. The options I know off are using `unwrap_or_else` or `expect` with `format!`
`unwrap_or_else` is lazy which is an advantage. Do you have a preference or a better suggestion? The other option is a static message in `expect` but that will not give you the exact file name thats causing the issue.

Reply:
Please check CI output: https://github.com/project-serum/anchor/runs/5412175525?check_suite_focus=true#step:8:340

Reply:
`cargo-fmt` / `clippy` are preferred
otherwise, we will need custom rules

Reply:
Okay. So im going to revert it back to the `unwrap_or_else` as suggested by `clippy` I know @armaniferrante suggested otherwise. I understand it looks a bit unwieldy. But it is better since it is lazy.